c_common/models/live_packet_gather/src/live_packet_gather.c

```diff
@@ -113,52 +114,54 @@ static struct lpg_config config;
 
 //! \brief Because WHY OH WHY would you use aligned memory? At least with this
 //! we don't get data aborts.
-static inline void write_unaligned(
-        void *base, uint32_t index, uint32_t value) {
+static inline void write_word(void *base, uint32_t index, uint32_t value) {
     uint16_t *ary = base;
     uint32_t idx = index * 2;
     ary[idx++] = CLAMP16(value);
     ary[idx] = CLAMP16(value >> 16);
 }
 
+//! \brief Simple mirror of write_word() for true 16 bit values.
+static inline void write_short(void *base, uint32_t index, uint32_t value) {
+    uint16_t *ary = base;
+    ary[index] = CLAMP16(value);
+}
+
+//! \brief Get how many events there are waiting to be sent
+static inline uint8_t get_event_count(void) {
+    uint8_t event_count = buffer_index;
+    // If there are payloads, it takes two buffer values to encode them
+    if (HAVE_PAYLOAD(config.packet_type)) {
+        event_count >>= 1;
+    }
+    return event_count;
+}
+
 static void flush_events(void) {
     // Send the event message only if there is data
-    if (buffer_index > 0) {
-        uint8_t event_count;
-
-        if ((config.packets_per_timestamp == 0)
-                || (packets_sent < config.packets_per_timestamp)) {
-            // Get the event count depending on if there is a payload or not
-            if (HAVE_PAYLOAD(config.packet_type)) {
-                event_count = buffer_index >> 1;
+    if ((buffer_index > 0) && (
+            (config.packets_per_timestamp == 0) ||
+            (packets_sent < config.packets_per_timestamp))) {
```

Comment:
It turns out that combining the two `if`s is now trivial.

Reply:
yep